### ISSUE TITLE
Add C# based websockets

### ISSUE THREAD

**Issue by csnewman** [csnewman/BlazorSignalR#13]
Currently websockets are only supported at a transport level, by ```WebSocketsTransport``` which is reimplemented in JS. This ticket tracks using the inbuilt ```WebSocketsTransport``` of ```SignalR.Client```

Need to check whether MonoWASM supports WebSockets yet.

**Comment by michaelp** [comment]
Just making sure that I got it right... You want to refactor websockets into an independent package so it would be used by the signalr client as a third party library. Correct?

**Comment by csnewman** [comment]
@michaelp Not quite. There is already an official websocket library ```System.Net.WebSockets```, managed by Microsoft. This issue tracks using that library once Mono WASM supports that library. (It may have already had support added since I last checked a few months ago.)
